ngraph/core/src/op/squeeze.cpp

```diff
@@ -18,23 +18,22 @@
 using namespace std;
 using namespace ngraph;
 
-NGRAPH_SUPPRESS_DEPRECATED_START
-
 NGRAPH_RTTI_DEFINITION(op::v0::Squeeze, "Squeeze", 0);
 
 op::Squeeze::Squeeze()
-    : FusedOp()
+    : Op()
 {
 }
 
 op::Squeeze::Squeeze(const Output<Node>& data, const Output<Node>& axes)
-    : FusedOp({data, axes})
+    : Op({data, axes})
 {
     constructor_validate_and_infer_types();
 }
 
-void op::Squeeze::pre_validate_and_infer_types()
+void op::Squeeze::validate_and_infer_types()
 {
+    NGRAPH_OP_SCOPE(v0_Squeeze_validate_and_infer_types);
     auto data = input_value(0);
```

Comment:
I think here we should check for element types in both inputs, specially axes input which is supposed to be just integer type. Additionally the second input is either scalar or 1D, I think we should also be able to add node validation check for this using compatible member function. Please add type_prop tests covering these cases.

Reply:
Added checks for 2nd input range and type.